### PR TITLE
Add pretty log format support, and hyperlinks for .bs files

### PR DESCRIPTION
**Pull request by georgejecook**
… files

### CODE REVIEW

File: src/BrightScriptFileUtils.ts [hunk 1]
```diff
@@ -1,13 +1,40 @@
...
+        } else if (fileName !== undefined && (fileName.toLowerCase().endsWith('.bs'))) {
+            return fileName.substring(0, fileName.length - 3) + '.xml';
         } else if (fileName !== undefined && fileName.toLowerCase().endsWith('.xml')) {
-            //jump to brs file
             return fileName.substring(0, fileName.length - 4) + '.brs';
         } else {
             return undefined;
         }
     }
+    public getAlternateBrsFileName(fileName: string): string | undefined {
```
Comment (TwitchBronBron) [inline src/BrightScriptFileUtils.ts:13]: Can you add a jsdoc comment for this method to explain its purpose and how it differs from `getAlternateFileName`?

File: src/LogOutputManager.ts [hunk 2]
```diff
@@ -278,6 +281,8 @@ export class LogOutputManager {
         let name = parts.length > 0 ? parts[parts.length - 1] : pkgPath;
         if (name.toLowerCase().endsWith('.xml') || name.toLowerCase().endsWith('.brs')) {
             name = name.substring(0, name.length - 4);
+        } else if (name.toLowerCase().endsWith('.xml')) {
```
Comment (TwitchBronBron) [inline src/LogOutputManager.ts:284]: We already handle .xml 2 lines previous. Was this supposed to be `.bs` ?

File: src/LogDocumentLinkProvider.ts [hunk 3]
```diff
@@ -99,13 +102,42 @@ export class LogDocumentLinkProvider implements vscode.DocumentLinkProvider {
...
-        pkgPath = util.removeFileScheme(pkgPath);
+        //remove preceeding pkg:
+        if (pkgPath.toLowerCase().indexOf('pkg:') === 0) {
+            pkgPath = pkgPath.substring(4);
+        }
 
         //use debugRootDir if provided, or rootDir if not provided.
-        let rootDir = this.launchConfig.debugRootDir ? this.launchConfig.debugRootDir : this.launchConfig.rootDir;
+        let rootDir = this.launchConfig.rootDir;
+        for (let i = 0; i < 2; i++) {
```
[the remark this answers is not in the document]
Comment (georgejecook) [reply src/LogDocumentLinkProvider.ts:112]: it's not random.. 
it ensures we call `pkgPath = fileUtils.getAlternateBrsFileName(pkgPath);` twice, so it calls this code for both pkgPath with bs and brs extension

File: src/LogDocumentLinkProvider.ts [hunk 4]
```diff
@@ -99,13 +102,42 @@ export class LogDocumentLinkProvider implements vscode.DocumentLinkProvider {
...
+                let clientPath = path.normalize(path.join(rootDir, pkgPath));
+                if (fsExtra.existsSync(clientPath)) {
+                    return clientPath;
+                }
+            }
 
-        let clientPath = path.normalize(path.join(rootDir, pkgPath));
-        return clientPath;
+            if (this.launchConfig.sourceDirs) {
+                if (this.launchConfig.sourceDirs.length === 1) {
```
[the remark this answers is not in the document]
Comment (georgejecook) [reply src/LogDocumentLinkProvider.ts:122]: I concur. fixed.

**Comment by TwitchBronBron** [comment]
@georgejecook you marked several comments as resolved, but I didn't see code changed or a comment for most of them. Did you perhaps forget to push the changes?

**Comment by georgejecook** [comment]
@TwitchBronBron pushed up my changes. soz for that.

**Comment by TwitchBronBron** [comment]
@georgejecook can you resolve the build errors?

**Comment by georgejecook** [comment]
done.

**Comment by georgejecook** [comment]
ok @TwitchBronBron should be all good now.

**Comment by georgejecook** [comment]
sad to say, that the latest I pushed up still has issues. 
image

there's some weird timing thing where links get out of sync with the log output. I've tried fixing this various ways over several hours, and can't throw any more time at it.

File: .vscode/settings.json [hunk 5]
```diff
@@ -17,5 +17,14 @@
     "editor.detectIndentation": false,
     "editor.tabSize": 4,
     "tslint.autoFixOnSave": true,
-    "typescript.format.insertSpaceAfterFunctionKeywordForAnonymousFunctions": false
+    "typescript.format.insertSpaceAfterFunctionKeywordForAnonymousFunctions": false,
```
[the remark this answers is not in the document]
Comment (TwitchBronBron) [reply .vscode/settings.json:20]: These are back again. Please remove the extra settings.

File: src/BrightScriptFileUtils.ts [hunk 6]
```diff
@@ -1,15 +1,23 @@
...
             return fileName.substring(0, fileName.length - 4) + '.brs';
-        } else {
-            return undefined;
         }
     }
+
+    /**
+     * Utility method to simplify getting alternate brs/bs extensions
+     * @param  {string} fileName
+     * @returns {string} Returns the brs file name, if filename is a brs file, otherwise the bs filename. If filenAMe is neither, then fileName is returned unchanged
```
Comment (TwitchBronBron) [inline src/BrightScriptFileUtils.ts:19]: Typo: `filenAMe`

**Comment by TwitchBronBron** [comment]
@georgejecook based on our latest conversation and the new [source literals PR in BrighterScript](https://github.com/rokucommunity/brighterscript/pull/126/files), I think you can remove a large portion of this PR now, right? Like the ProjectManager/SourceManager stuff.

**Comment by georgejecook** [comment]
added the spelling correction you requested

**Comment by georgejecook** [comment]
superceded by https://github.com/rokucommunity/vscode-brightscript-language/pull/262

**Pull request closed without merging**